test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs

```diff
@@ -0,0 +1,344 @@
+            Assert.Equal(1, configuredSources.Count);
+            Assert.Equal("source1", configuredSources.First());
+        }
+
+        [Fact]
+        public void SearchTree_TopNodeIsNotGlobbing()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <packageSource key=""source1"">
+            <namespace id=""NuGet.Common"" />
+        </packageSource>
+        <packageSource key=""source2"">
+            <namespace id=""NuGet"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.IsNamespacesEnabled);
+            configuration.Namespaces.Should().HaveCount(2);
+
+            // Since previous node is not globbing it shouldn't match anything.
+            var configuredSources = configuration.GetConfiguredPackageSources("NuGet.Common1");
+            Assert.Equal(null, configuredSources);
+
+            configuredSources = configuration.GetConfiguredPackageSources("NuGet");
+            Assert.Equal(1, configuredSources.Count);
+            Assert.Equal("source2", configuredSources.First());
+
+            configuredSources = configuration.GetConfiguredPackageSources("NuGet.Common");
+            Assert.Equal(1, configuredSources.Count);
+            Assert.Equal("source1", configuredSources.First());
+        }
+
+        [Fact]
+        public void SearchTree_TopBottomNodesBothGlobbing()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <packageSource key=""source1"">
+            <namespace id=""NuGet.*"" />
+        </packageSource>
+        <packageSource key=""source2"">
+            <namespace id=""NuGet.Common.*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.IsNamespacesEnabled);
+            configuration.Namespaces.Should().HaveCount(2);
+
+            var configuredSources = configuration.GetConfiguredPackageSources("NuGet.Common1");
```

Comment:
you have no tests that check for `source2`

Reply:
https://github.com/NuGet/NuGet.Client/blob/42e68a74fce207a2b9c442886071f51d2c20ebe5/test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs#L295-L297